Bottom: a47a7032a5f472423d9529ebe8e2924bac63f996
Top:    0078d209db0092f83fe8f7e0c1e3b29832f3ef67
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-05 22:42:07 +0100

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index d5d08f1..feed3b6 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -306,6 +306,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
